src/main/java/cloud/orbit/actors/extensions/metrics/dropwizard/InstrumentedMessageSerializer.java

```diff
@@ -0,0 +1,61 @@
+package cloud.orbit.actors.extensions.metrics.dropwizard;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+
+import cloud.orbit.actors.extensions.MessageSerializer;
+import cloud.orbit.actors.runtime.BasicRuntime;
+import cloud.orbit.actors.runtime.Message;
+import cloud.orbit.actors.runtime.MessageDefinitions;
+
+public class InstrumentedMessageSerializer implements MessageSerializer {
+    private final MetricRegistry metricRegistry;
+    private final MessageSerializer messageSerializer;
+    private Map<Integer, Histogram> serializeMetrics;
+    
+    public InstrumentedMessageSerializer(MessageSerializer messageSerializer) {
+        this(new MetricRegistry(), messageSerializer);
+    }
+
+    public InstrumentedMessageSerializer(MetricRegistry metricRegistry, MessageSerializer messageSerializer)
+    {
+        this.metricRegistry = metricRegistry;
+        this.messageSerializer = messageSerializer;
+        setupMetrics();
+    }
+    
+    public MetricRegistry getMetricRegistry() {
+        return metricRegistry;
+    }
+
+    private void setupMetrics()

```

Comment:
I know this class is metric-implementation agnostic, but valid Datadog metric names cannot contain hyphens.
https://docs.datadoghq.com/faq/#what-are-valid-metric-names

Peeking at the other classes, it looks like we're using underscores elsewhere.